packages/sources/tradingeconomics/src/config.ts

```diff
@@ -11,27 +13,65 @@ import { util } from '@chainlink/ea-bootstrap'
  *      required: true
  *    API_CLIENT_SECRET:
  *      required: true
- *    SYMBOLS:
- *      required: true
- *    RECONNECT_TIMEOUT:
- *      required: false
- *      default: 3000
  */
 
-export type Config = {
-  url: string
-  key: string
-  secret: string
-  symbols: string
-  reconnectTimeout: number
+export type Config = config & {
+  client: {
+    key: string
+    secret: string
+  }
 }
 
+export const NAME = 'TRADINGECONOMICS'
+
+export const DEFAULT_API_ENDPOINT = 'https://api.tradingeconomics.com/markets'
+const DEFAULT_WS_API_ENDPOINT = 'ws://stream.tradingeconomics.com/'
+
 export const makeConfig = (prefix?: string): Config => {
+  const config = Requester.getDefaultConfig(prefix)
+  config.api.baseURL = config.api.baseURL || DEFAULT_API_ENDPOINT
+
   return {
-    url: util.getEnv('API_URL', prefix) || 'ws://stream.tradingeconomics.com/',
-    key: util.getRequiredEnv('API_CLIENT_KEY', prefix),
-    secret: util.getRequiredEnv('API_CLIENT_SECRET', prefix),
-    symbols: util.getRequiredEnv('SYMBOLS', prefix),
-    reconnectTimeout: Number(util.getEnv('RECONNECT_TIMEOUT', prefix) || 3000),
+    ...config,
+    client: {
+      key: util.getRequiredEnv('API_CLIENT_KEY', prefix),
+      secret: util.getRequiredEnv('API_CLIENT_SECRET', prefix),
+    },
+  }
+}
+
+export const makeWSHandler = (config?: Config): MakeWSHandler => {
+  // http://api.tradingeconomics.com/documentation/Streaming
+  // https://github.com/boxhock/tradingeconomics-nodejs-stream/blob/master/src/index.ts
+  const withApiKey = (url: string, key: string, secret: string) => `${url}?client=${key}:${secret}`
+  const getSubscription = (to: string) => ({ topic: 'subscribe', to })
+
+  return () => {
+    const defaultConfig = config || makeConfig()
+
+    return {
+      connection: {
+        url: withApiKey(
+          defaultConfig.api.baseWsURL || DEFAULT_WS_API_ENDPOINT,
+          defaultConfig.client.key || '',
+          defaultConfig.client.secret || '',
+        ),
+      },
+      subscribe: input => {
+        const validator = new Validator(input, customParams)
+        if (validator.error) {
+          return
+        }
+        const base = validator.overrideSymbol(NAME).toUpperCase()
+        return getSubscription(base)
+      },
+      unsubscribe: () => '',
```

Comment:
Does the whole connection need to be torn down to unsubscribe? I don't understand these API providers and why they don't handle unsubscribe messages.

Reply:
Yeah...I couldn't find an unsubscribe command in their SDK:
https://github.com/tradingeconomics/tradingeconomics/tree/master/nodejs/stream-nodejs

or in the one that Jonas made either :(

Reply:
Ok, then make this `undefined` when Rodrigo makes the underlying changes.